### ISSUE TITLE
Testresults are not reported when using Gitlab Matrix Jobs

### ISSUE THREAD

**Issue by FlorianBorn** [MicroFocus/octane-gitlab-service#105]
Hello,

we just noticed that the octane-gitlab-service fails to report test results when the results are generated within gitlab matrix jobs.

Here are the logs:
<details>
<summary>logs</summary>

```
[WARN ] 2023-12-05 15:09:50.277 [TestsResultPushWorker-65] OctaneServices - Failed to return test results
org.gitlab4j.api.GitLabApiException: 404 Project Not Found
	at org.gitlab4j.api.AbstractApi.validate(AbstractApi.java:678) ~[gitlab4j-api-5.2.0.jar!/:?]
	at org.gitlab4j.api.AbstractApi.get(AbstractApi.java:214) ~[gitlab4j-api-5.2.0.jar!/:?]
	at org.gitlab4j.api.JobApi.getJob(JobApi.java:204) ~[gitlab4j-api-5.2.0.jar!/:?]
	at com.microfocus.octane.gitlab.services.OctaneServices.getTestsResult(OctaneServices.java:382) ~[classes!/:23.3.2]
	at com.hp.octane.integrations.services.tests.TestsServiceImpl.doPreflightAndPushTestResult(TestsServiceImpl.java:293) ~[integrations-sdk-2.7.9.1.jar!/:?]
	at com.hp.octane.integrations.services.tests.TestsServiceImpl.worker(TestsServiceImpl.java:270) ~[integrations-sdk-2.7.9.1.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:840) ~[?:?]
[INFO ] 2023-12-05 15:09:50.282 [TestsResultPushWorker-65] TestsServiceImpl - Using base64, https://octane.dummy.de/internal-api/shared_spaces/5001/analytics/ci/servers/octane-testautomation/jobs/dGVzdGF1dG9tYXRpb24vdG9vbHMvcm9ib3RmcmFtZXdvcmsvcm9ib3RfYXBwaXVtL1BPQ19BcHBpdW06IFstaSBsb2NhbC1icm93c2VyIC1lIFRvRG8sYmxvY2tlZCAtLWxvZ2xldmVsIFRSQUNFOklORk8gLS1saXN0ZW5lciAkQVJUSUZBQ1RTX0RJUi8kQVJUSUZBQ1RTX1JPQk9UX09VVFBVVF9ESVIgLXggeHVuaXQueG1sIC0tb3V0cHV0ZGlyICRBUlRJRkFDVFNfRElSLyRBUlRJRkFDVFNfUk9CT1RfT1VUUFVUX0RJUl0=/tests-result-preflight?ci-job-encoding=base64
[INFO ] 2023-12-05 15:09:50.398 [TestsResultPushWorker-65] TestsServiceImpl - [https://octane.dummy.de?p=5001] test results preflight 'replaced/by/dummy: [-i local-browser -e ToDo,blocked --loglevel TRACE:INFO --listener $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR -x xunit.xml --outputdir $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR] #173132010 = true
[ERROR] 2023-12-05 15:09:50.622 [TestsResultPushWorker-65] TestsServiceImpl - [https://octane.dummy.de?p=5001] permanent error on 'replaced/by/dummy: [-i local-browser -e ToDo,blocked --loglevel TRACE:INFO --listener $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR -x xunit.xml --outputdir $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR] #173132010, passing over
com.hp.octane.integrations.exceptions.PermanentException: push request PERMANENTLY failed with status 400
	at com.hp.octane.integrations.services.tests.TestsServiceImpl.doPreflightAndPushTestResult(TestsServiceImpl.java:333) ~[integrations-sdk-2.7.9.1.jar!/:?]
	at com.hp.octane.integrations.services.tests.TestsServiceImpl.worker(TestsServiceImpl.java:270) ~[integrations-sdk-2.7.9.1.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:840) ~[?:?]
```
</details>

Here is an example pipeline to reproduce the error:
```
---
stages:
  - "dummy"

variables:
    WORKSPACE: "./"

create_report: 
    stage: "dummy"
    when: on_success
    cache: {}
    script:
        - |
          echo '<?xml version="1.0" encoding="UTF-8"?>
          <testsuites time="15.682687">
              <testsuite name="Tests.Registration" time="6.605871">
                  <testcase name="testCase1" classname="Tests.Registration" time="2.113871" />
                  <testcase name="testCase2" classname="Tests.Registration" time="1.051" />
                  <testcase name="testCase3" classname="Tests.Registration" time="3.441" />
              </testsuite>
              <testsuite name="Tests.Authentication" time="9.076816">
                  <testsuite name="Tests.Authentication.Login" time="4.356">
                      <testcase name="testCase4" classname="Tests.Authentication.Login" time="2.244" />
                      <testcase name="testCase5" classname="Tests.Authentication.Login" time="0.781" />
                      <testcase name="testCase6" classname="Tests.Authentication.Login" time="1.331" />
                  </testsuite>
                  <testcase name="testCase7" classname="Tests.Authentication" time="2.508" />
                  <testcase name="testCase8" classname="Tests.Authentication" time="1.230816" />
                  <testcase name="testCase9" classname="Tests.Authentication" time="0.982">
                      <failure message="Assertion error message" type="AssertionError">
                          <!-- Call stack printed here -->
                      </failure>            
                  </testcase>
              </testsuite>
          </testsuites>' > result.octane.xml

    parallel:
      matrix:
        - RUN_OPTIONS: "-i local-browser -e ToDo,blocked --loglevel TRACE:INFO --listener $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR -x xunit.xml --outputdir $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR"
        - RUN_OPTIONS: "-e local-browser,ToDo,blocked --loglevel TRACE:INFO --listener $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR -x xunit.xml --outputdir $ARTIFACTS_DIR/$ARTIFACTS_ROBOT_OUTPUT_DIR"        
    artifacts:
        when: always
        paths:
          - $WORKSPACE/**/*.octane.xml
```
Without the parallel/matrix part, the test results are reported as expected.

We are using Version: 23.3.2

**Comment by TiutiuMadalin** [comment]
Hello @FlorianBorn, thank you for the detailed logs and pipeline example,  I managed to reproduce and I think I found the issue and a fix for it.
The Issue was coming from a path not being correctly parsed because the job name contained multiple "/" it could not parse it.
![image](https://github.com/MicroFocus/octane-gitlab-service/assets/62649591/4c8fb475-ff04-485c-968a-71f4f0cd1045)
![image](https://github.com/MicroFocus/octane-gitlab-service/assets/62649591/a3df8d04-eda2-4885-b6b6-7fb0d0d8cbca)

**Comment by TiutiuMadalin** [comment]
fixed with [v23.4.0-CE](https://github.com/MicroFocus/octane-gitlab-service/releases/tag/v23.4.0-CE)
